_docs/concepts/security/authn-policy.md

```diff
@@ -0,0 +1,101 @@
+
+{% include figure.html width='80%' ratio='100%'
+    img='./img/authn.png'
+    alt='Istio authentication policy architecture'
+    title='Istio authentication policy architecture'
+    caption='Istio authentication policy architecture'
+    %}
+
+
+
+Policy is scoped at namespace level, with (optional) target selector rules to narrow down the set of services (within the same namespace as the policy) on which the policy should be applied. This aligns with the ACL model based on Kubernetes RBAC. More specifically, only admin of the namespace can set policies for services in that namespace.
+
+
+Authentication engine is implemented on sidecars. For example, with Envoy sidecar, it is a combination of SSL settings and HTTP filters. If authentication fails, requests will be rejected (either with SSL handshake error code, or http 401, depending on the type of authencation mechanism). If authentication succeeds, the following authenticated attributes will be generated:
+
+- **source.principal**: peer principal. If peer authentiation is not used, the attribute is not set.
+- **request.auth.principal**: depends on the policy principal binding, this could be peer principal (if USE_PEER) or origin principal (if USE_ORIGIN).
+- **request.auth.audience**: reflect the audience (`aud`) claim within the origin-JWT (JWT that is used for origin authentication)
+- **request.auth.presenter**: similarly, reflect the authorized presenter (`azp`) claim of the origin-JWT.
+- **request.auth.claims**: all raw string claims from origin-JWT.
+
+Origin principal is not explicitely output. In general, it can always be reconstructed by joining (`iss`) and subject (`sub`) claims with "/" seperator (for example, if `iss` and `sub` claims are "*googleapis.com*" and "*123456*" respectively, then origin principal is "*googleapis.com/123456*"). On the other hand, if principal binding is USE_ORIGIN, **request.auth.principal** carries the same value as origin principal.
+
+
+## Anatomy of the policy
+
+### Target selectors
+
+Defines rule to find service(s) on which policy should be applied. If no rule provided, the policy will be matched to all services in the namespace, so call namespace-level policy (as opposed to service-level policies which have non-empty selector rules). Istio (pilot) will pick the service-level policy if available, otherwise fallback to namespace-level policy. If neither is defined, it uses the default policy based on service mesh config and/or service annotation, which can only set mutual TLS setting (these are pre-0.7 mechanisms to config mutual TLS for Istio service mesh). See [testing Istio nutual TLS]({{home}}/docs/tasks/security/mutual-tls.html) and [per-service mutual TLS enablement]({{home}}/docs/tasks/security/per-service-mtls.html) for more details.
+
+Operators are responsible for avoiding conflicts, e.g create more than one service-level policy that match to the same service(s) (or more than one namespace-level policy on the same namespace).
```

Comment:
How do we resolve it if we see such conflicts? Randomly pick one? Or return any validation failure?

Reply:
More or lest random at the moment (or whatever the order of the all CRDs when we request). In any case, there is no guarantee what will be picked, at least at this point so I think it simpler to make customer aware of that.

About validation, ideally, we should check for conflict when policy is submitted (via istioctl). However, the current framework allows us to look at the new one only. It could be something we can improve in the near future.

Reply:
Understood. I was just wondering if we need to document the consequence of policy conflict (e.g., if there is a conflict, there is no guarantee which policy is picked). But that's a minor point. I will leave it to you to decide.